tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CollectInformation.scala

```diff
@@ -125,102 +120,16 @@ class CollectInformation(apps: Seq[ApplicationInfo],
-              mean(col("duration")).alias("mean_dur"),
-              count(col("duration")).alias("num_tasks"))
-            .collect().map { row =>
-          val stageId = row.getInt(0)
-          val minDur = row.getLong(1)
-          val maxDur = row.getLong(2)
-          val meanDur = row.getDouble(3)
-          val numTasks = row.getLong(4)
-          (stageId, StageMetrics(numTasks.toInt,
-            s"MIN: ${formatter.format(minDur)} ms " +
-                s"MAX: ${formatter.format(maxDur)} ms " +
-                s"AVG: ${formatter.format(meanDur)} ms"))
-        }.toMap
-
-        val accumSummary = accums
-          .select(col("sqlId"), col("accumulatorId"), col("max_value"))
-          .collect()
-        val sqlIdToMaxMetric = new mutable.HashMap[Long, ArrayBuffer[(Long,Long)]]()
-        for (row <- accumSummary) {
-          val list = sqlIdToMaxMetric.getOrElseUpdate(row.getLong(0),
-            new ArrayBuffer[(Long, Long)]())
-          list += row.getLong(1) -> row.getLong(2)
-        }
-
-        val sqlPlansMap = app.sqlPlan.map { case (sqlId, sparkPlanInfo) =>
-          sqlId -> ((sparkPlanInfo, app.physicalPlanDescription(sqlId)))
-        }
-        for ((sqlID,  (planInfo, physicalPlan)) <- sqlPlansMap) {
-          val dotFileWriter = new ToolTextFileWriter(outputDirectory,
-            s"${app.appId}-query-$sqlID.dot")
-          try {
-            val metrics = sqlIdToMaxMetric.getOrElse(sqlID, Seq.empty).toMap
-            GenerateDot.writeDotGraph(QueryPlanWithMetrics(planInfo, metrics),
-              physicalPlan, accumIdToStageId, stageIdToStageMetrics, dotFileWriter,
-              sqlID, app.appId)
-          } finally {
-            dotFileWriter.close()
-          }
-        }
-
-        val duration = TimeUnit.SECONDS.convert(System.nanoTime() - start, TimeUnit.NANOSECONDS)
-        fileWriter.foreach(_.write(s"Generated DOT graphs for app ${app.appId} " +
-          s"to $outputDirectory in $duration second(s)\n"))
-      } else {
-        val missingDataFrames = requiredDataFrames.filterNot(app.allDataFrames.contains)
-        fileWriter.foreach(_.write(s"Could not generate DOT graph for app ${app.appId} " +
-          s"because of missing data frames: ${missingDataFrames.mkString(", ")}\n"))
-      }
-    }
-  }
-
   // Print SQL Plan Metrics
-  def printSQLPlanMetrics(shouldGenDot: Boolean, outputDir: String): Unit = {
+  def printSQLPlanMetrics(): Unit = {
     for (app <- apps){
       if (app.allDataFrames.contains(s"sqlMetricsDF_${app.index}") &&
         app.allDataFrames.contains(s"driverAccumDF_${app.index}") &&
         app.allDataFrames.contains(s"taskStageAccumDF_${app.index}") &&
         app.allDataFrames.contains(s"jobDF_${app.index}") &&
         app.allDataFrames.contains(s"sqlDF_${app.index}")) {
         val messageHeader = "\nSQL Plan Metrics for Application:\n"
-        val accums = app.runQuery(app.generateSQLAccums, fileWriter = fileWriter,
```

Comment:
so we lost the reuse of generateSQLAccums here, not sure how long that takes but since its sql query probably non-trivial we may want to see if we can generate once and reuse

Reply:
That generic reuse did not work.  It was not caching the data so the only thing it saved on was re-parsing the SQL query into a data frame.

Reply:
If you want me to I can make the data returned a lazy value in ApplicationInfo so it does end up being shared.

Reply:
ah right, we change the query in generate dot so don't get any stage reuse